recipes/json_dto/all/conanfile.py

```diff
@@ -16,51 +21,52 @@ class JsondtoConan(ConanFile):
     no_copy_source = True
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    def requirements(self):
-        self.requires("rapidjson/1.1.0")
+    def _min_cppstd(self):
+        return 14
 
-    def validate(self):
-        minimal_cpp_standard = "14"
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, minimal_cpp_standard)
-        minimal_version = {
+    @property
+    def _compilers_minimum_version(self):
+        return {
             "gcc": "5",
             "clang": "4",
             "apple-clang": "8",
             "Visual Studio": "15"
         }
-        compiler = str(self.settings.compiler)
-        if compiler not in minimal_version:
-            self.output.warn(
-                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
-            self.output.warn(
-                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
-        elif tools.Version(self.settings.compiler.version) < minimal_version[compiler]:
-            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
 
-        if self.settings.compiler == "apple-clang" and tools.Version(self.settings.compiler.version) >= "11":
-            raise ConanInvalidConfiguration(f"{self.name} requires apple-clang less then version 11")
+    def validate(self):
```

Comment:
move `validate()` after `package_id()` please to follow conan order of execution, otherwise it's hard to reason.

Reply:
done